server.js

```diff
@@ -1,103 +1,189 @@
+}
+
+function initialiseProxy (app) {
+  app.enable('trust proxy')
+}
+
+function initialiseCookies (app) {
+  cookies.configureSessionCookie(app)
 }
-app.use(/\/((?!images|public|stylesheets|javascripts).)*/, loggingMiddleware(
-      ':remote-addr - :remote-user [:date[clf]] ":method :url HTTP/:http-version" :status :res[content-length] ":referrer" ":user-agent" - total time :response-time ms'))
-
-app.use(i18n.init)
-app.use(compression())
-app.use(staticify.middleware)
-
-app.enable('trust proxy')
-app.disable('x-powered-by')
-
-cookies.configureSessionCookie(app)
-
-app.engine('html', require(path.join(__dirname, '/lib/template-engine.js')).__express)
-app.set('view engine', 'html')
-app.set('vendorViews', path.join(__dirname, '/govuk_modules/govuk_template/views/layouts'))
-app.set('views', path.join(__dirname, '/app/views'))
-
-app.use(bodyParser.json())
-app.use(bodyParser.urlencoded({extended: true}))
-
-app.use('/javascripts', express.static(path.join(__dirname, '/public/assets/javascripts'), publicCaching))
-app.use('/images', express.static(path.join(__dirname, '/public/images'), publicCaching))
-app.use('/stylesheets', express.static(path.join(__dirname, '/public/assets/stylesheets'), publicCaching))
-
-app.use('/public', express.static(path.join(__dirname, '/public'), publicCaching))
-app.use('/public', express.static(path.join(__dirname, '/govuk_modules/govuk_template/assets'), publicCaching))
-app.use('/public', express.static(path.join(__dirname, '/govuk_modules/govuk_frontend_toolkit'), publicCaching))
-
-app.use(favicon(path.join(__dirname, 'govuk_modules', 'govuk_template', 'assets', 'images', 'favicon.ico')))
-app.use(function (req, res, next) {
-  res.locals.assetPath = '/public/'
-  if (typeof process.env.ANALYTICS_TRACKING_ID === 'undefined') {
-    logger.warn('Google Analytics Tracking ID [ANALYTICS_TRACKING_ID] is not set')
-    res.locals.analyticsTrackingId = '' // to not break the app
-  } else {
-    res.locals.analyticsTrackingId = process.env.ANALYTICS_TRACKING_ID
+
+function initialiseTemplateEngine (app) {
+  // Configure nunjucks
+  // see https://mozilla.github.io/nunjucks/api.html#configure
+  const nunjucksConfiguration = {
+    express: app, // the express app that nunjucks should install to
+    autoescape: true, // controls if output with dangerous characters are escaped automatically
+    throwOnUndefined: false, // throw errors when outputting a null/undefined value
+    trimBlocks: true, // automatically remove trailing newlines from a block/tag
+    lstripBlocks: true, // automatically remove leading whitespace from a block/tag
+    watch: false, // reload templates when they are changed (server-side). To use watch, make sure optional dependency chokidar is installed
+    noCache: false // never use a cache and recompile templates each time (server-side)
   }
-  res.locals.session = function () {
-    return session.retrieve(req, req.chargeId)
+  if ((!NODE_ENV) || (NODE_ENV !== 'production')) {
```

Comment:
This could be inlined;
```
watch: NODE_ENV !== 'production',
noCache: NODE_ENV !== 'production'
```

That means you could inline both `APP_VIEWS` and `nunjucksConfiguration`

Reply:
Thanks, Max.

I declared the `if ((!NODE_ENV) || (NODE_ENV !== 'production'))` block in case we don't have `NODE_ENV` defined (in dev using `msl`) and also in case we may need other `non-boolean` properties for non-prod declarations as in https://github.com/alphagov/pay-products-ui/blob/d157974c7ddacb834b9f43e6e0e31653e9544652/server.js#L83.
I can change it like:
```
watch: ((!NODE_ENV) || (NODE_ENV !== 'production')),
noCache: ((!NODE_ENV) || (NODE_ENV !== 'production'))
```
adding extra parentheses for more clarity.
Let me know what you prefer.